pandas/core/frame.py

```diff
@@ -4323,7 +4323,8 @@ def infer(x):
     # ----------------------------------------------------------------------
     # Merging / joining methods
 
-    def append(self, other, ignore_index=False, verify_integrity=False):
+    def append(self, other, ignore_index=False, verify_integrity=False,
+               dropna=True):
```

Comment:
Sorry I was wrong about it. The removal of the `dropna` fails test `pandas.tools.tests.test_pivot.test_margin_dropna`. When margins are appended, the missing of `dropna` leads to the drop of `NaN` level. This could also be solved by specifying the `dropna` parameter in `MultiIndex` class constructor.